plotly/plotly/plotly.py

```diff
@@ -1308,7 +1308,19 @@ def response_handler(cls, response):
                     .format(url=get_config()['plotly_api_domain'])
                 )
             else:
-                raise requests_exception
+                try:
+                    parsed_response = response.json()
+                except:
+                    parsed_response = response.content
+
+                try:
+                    # try get a friendly error message
+                    errors = parsed_response.get('errors')
+                    message = errors[-1]['message']
+                except:
+                    raise requests_exception
+
+                raise exceptions.PlotlyError(message)
```

Comment:
> Want to try and follow that more closely, OR, extract the common functionality out? It'd be great to just have one error handler.

that's what I was hoping to achieve, I took the main logic and moved it to the response_handler and use it in animations to catch errors instead of having a code block in animations for that.

> In particular, the code you wrote assume that errors is not empty, which we might not want to assume

it's in a `try` bloc, so if my assumption is incorrect then it raises the usual `requests_exception`. I could put two `if` for looking to see if `errors` and `message` exist but would that not do the same?